peer_client.go

```diff
@@ -170,8 +174,12 @@ func (c *PeerClient) UpdatePeerGlobals(ctx context.Context, r *UpdatePeerGlobals
 	}
 
 	// See NOTE above about RLock and wg.Add(1)
+	c.mutex.RLock()
 	c.wg.Add(1)
-	defer c.wg.Done()
+	defer func() {
+		c.mutex.RUnlock()
+		defer c.wg.Done()
+	}()
```

Comment:
Could this be simpler as:
```
defer c.wg.Done()
defer c.mutex.RUnlock()
```

Reply:
It then becomes 2 defers instead of one defer.